.github/workflows/release.yml

```diff
@@ -1,86 +1,100 @@
         uses: actions/checkout@v2
+
       - name: Install toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
+
       - uses: Swatinem/rust-cache@v1
         with:
           cache-on-failure: true
-      - name: cargo build
+
+      - name: Build binaries
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release
-
-      - name: Upload production artifacts
-        uses: actions/upload-artifact@v2
-        with:
-            name: target-${{ matrix.os }}
-            path: |
-              ./target/release/forge
-              ./target/release/cast
+          args: --release --bins
 
-  # creates a GH release with the built binaries and CHANGELOG entry
-  create-release:
-    runs-on: ubuntu-latest
-    needs:  build-artifacts
-    steps:
-     - name: Checkout
-       uses: actions/checkout@v2
-       with:
-        fetch-depth: 1000
-
-     - name: Set output
-       id: vars
-       run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
-
-     - name: Restore artifacts
-       uses: actions/download-artifact@v2
+      - name: Strip binaries (non-Windows)
+        if: matrix.job.platform != 'win32'
+        run: |
+          cd target/release
+          strip cast
+          strip forge
 
-     - name: Build Changelog
-       id: github_release
-       uses: mikepenz/release-changelog-builder-action@v2.4.2
-       env:
-        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
-
-     - name: Create Release
-       id: create-release
-       uses: softprops/action-gh-release@v0.1.13
-       env:
-        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
-       with:
-         tag_name:  ${{ steps.vars.outputs.tag }}
-         release_name: ${{ github.ref }}
-         body: ${{steps.build_changelog.outputs.changelog}}
-         files: |
-           ./target-macos-latest
-           ./target-ubuntu-latest
+      - name: Archive binaries
+        id: artifacts
+        env:
+          PLATFORM_NAME: ${{ matrix.job.platform }}
+          ARCH: ${{ matrix.job.arch }}
+        run: |
+          if [ "$PLATFORM_NAME" != "win32" ]; then
+            tar -czvf "foundry_${TAG_NAME}_${PLATFORM_NAME}_${ARCH}.tar.gz" -C ./target/release forge cast
+            echo "::set-output name=file_name::foundry_${TAG_NAME}_${PLATFORM_NAME}_${ARCH}.tar.gz"
+          else
+            cd ./target/release
+            7z a -tzip "foundry_${TAG_NAME}_${PLATFORM_NAME}_${ARCH}.zip" forge.exe cast.exe
+            mv "foundry_${TAG_NAME}_${PLATFORM_NAME}_${ARCH}.zip" ../../
+            echo "::set-output name=file_name::foundry_${TAG_NAME}_${PLATFORM_NAME}_${ARCH}.zip"
+          fi
+        shell: bash
+ 
+      - name: Build changelog
+        id: build_changelog
+        if: ${{ env.IS_NIGHTLY == false }}
+        uses: mikepenz/release-changelog-builder-action@v2.4.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```

Comment:
To make this work I need to issue and add a `GH_TOKEN` secret right?

Reply:
Hmm, I just checked the docs and it does automatically inject a `GITHUB_TOKEN`, not sure why this worked without me adding a `GH_TOKEN` secret. I'll investigate

Reply:
Ok so this was a mistake but it didn't fail :shrug: Fixed it to use the token that is automatically injected into the runner